registration_api.js

```diff
@@ -0,0 +1,694 @@
+    const signup_device = signup_device_cookie.get('signup_device');
+    const date_first_contact_cookie = new CookieStorage('date_first_contact');
+    const date_first_contact = date_first_contact_cookie.get('date_first_contact');
+    const marketing_queries = `${signup_device ? `&signup_device=${signup_device}` : ''}${date_first_contact ? `&date_first_contact=${date_first_contact}` : ''
+        }`;
+    const getOAuthUrl = () => {
+        return `https://oauth.deriv.com/oauth2/authorize?app_id=${getSessionAppId()}&l=${language}${marketing_queries}&brand=deriv`;
+    };
+
+    if (server_url && /qa/.test(server_url)) {
+        return `https://${server_url}/oauth2/authorize?app_id=${getSessionAppId()}&l=${language}${marketing_queries}&brand=deriv`;
+    }
+    return getOAuthUrl();
+};
+
+function activate(state) {
+    const joinedState = state.toStrings().join(' ');
+    const elApp = document.getElementById('app-registration-machine');
+    const form_checkbox = document.getElementById('expand_form');
+    const is_folded_state = joinedState === 'logged_in logged_in.register_tab logged_in.register_tab.folded_form';
+    if (elApp) {
+        elApp.dataset.state = joinedState;
+        elApp.setAttribute("data-state", joinedState);
+    }
+    if (is_folded_state) {
+        if (form_checkbox) form_checkbox.checked = false;
+    } else {
+        if (form_checkbox) form_checkbox.checked = true;
+    }
+}
+
+const interpreter = XState
+    .interpret(appRegistrationMachine)
+    .onTransition(activate)
+    .start(sessionState);
+
+const { send } = interpreter;
+
+const unfolded_form_checkbox = document.getElementById('expand_form');
+if (unfolded_form_checkbox) {
+    unfolded_form_checkbox.addEventListener('change', () => {
+        send({
+            "type": "TOGGLE_FORM"
+        });
+    });
+}
+
+const register_button = document.getElementById('register_button');
+if (register_button) {
+    register_button.addEventListener('click', () => {
+        send({
+            "type": "REGISTER_TOGGLE_TAB"
+        });
+    });
+}
+
+const manage_button = document.getElementById('manage_button');
+if (manage_button) {
+    manage_button.addEventListener('click', () => {
+        send({
+            "type": "MANAGE_TOGGLE_TAB"
+        });
+    });
+}
+
+const logout_button = document.getElementById('logout_button');
+if (logout_button) {
+    logout_button.addEventListener('click', () => {
+        send({
+            "type": "LOGOUT"
+        });
+    });
+}
+
+const registerLoginButton = document.getElementById('registerLogin');
+if (registerLoginButton) {
+    registerLoginButton.addEventListener('click', () => {
+        redirectToLogin(false, 'EN');
+    });
+};
+
+const getAppList = async () => {
+    const app_id = getSessionAppId();
+    const endpoint = getEndpoint();
+    const api = new DerivAPIBasic({ endpoint, lang: 'EN', app_id });
+    const token1 = getToken();
+    await api.authorize(token1);
```

Comment:
I see a few authorize calls here. Do we have a handling for if the token is not able to be authorized?

Reply:
good catch. There is possible error that is unhandled here. Will focus on addressing it in UI